code/iaas/logic/src/main/resources/META-INF/cattle/core-process/spring-core-processes-context.xml

```diff
@@ -18,15 +18,15 @@
         <process:defaultProcesses resourceType="agent" />
 
         The above XML is the equivalent as doing the below, 
-        refer to cattle-process-common/src/main/resources/META-INF/cattle/process/defaults.properties for the definitive list
+        refer to cattle/code/iaas/logic-common/src/main/resources/META-INF/cattle/process/defaults.properties for the definitive list
 
-        TODO: This is out of date
-
-        <process:process name="agent.create" resourceType="agent" start="requested" transitioning="registering" done="inactive" />
-        <process:process name="agent.activate" resourceType="agent" start="inactive" transitioning="activating" done="active" />
-        <process:process name="agent.deactivate" resourceType="agent" start="active,activating" transitioning="deactivating" done="inactive" />
-        <process:process name="agent.remove" resourceType="agent" start="active,activating,deactivating,registering" transitioning="removing" done="removed" />
+        <process:process name="agent.create" resourceType="agent" start="requested" transitioning="registering" done="inactive" delegate="agent.activate" />
+        <process:process name="agent.activate" resourceType="agent" start="inactive,registering" transitioning="activating" done="active" />
+        <process:process name="agent.deactivate" resourceType="agent" start="active,activating,updating-active,updating-inactive" transitioning="deactivating" done="inactive" delegate="agent.remove" />
+        <process:process name="agent.remove" resourceType="agent" start="requested,inactive,activating,deactivating,registering,updating-active,updating-inactive" transitioning="removing" done="removed" />
         <process:process name="agent.purge" resourceType="agent" start="removed" transitioning="purging" done="purged" />
+        <process:process name="agent.update" resourceType="agent" start="inactive,active" transitioning="inactive=updating-inactive,active=updating-active" done="updating-inactive=inactive,updating-active=active" />
```

Comment:
I actually didn't know you could do this compound syntax where you end up having multiple done states in one process definition. Do you have an xml example you based this off of?